src/service/xhr-impl.js

```diff
@@ -124,9 +122,9 @@ export class Xhr {
     // Do not append __amp_source_origin if explicitly disabled
     // (requireAmpResponseSourceOrigin overrides disable).
     dev().assert(
-      !(init.disableAmpSourceOrigin && init.requireAmpResponseSourceOrigin),
+      !(init.ampCors && init.requireAmpResponseSourceOrigin),
```

Comment:
in case ampCors == false, let's just override the value of requireAmpResponseSourceOrigin to false.